### PR TITLE
feat: teach ChunkedLayout how to read metadata

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

File: vortex-file/src/read/layouts/chunked.rs [hunk 1]
```diff
@@ -111,11 +140,15 @@ impl ChunkedLayout {
         self.children()
             .zip_eq(self.child_ranges())
             .map(|((i, c), (begin, end))| {
+                let layout_part_id = i as u16;
+                if layout_part_id == METADATA_LAYOUT_PART_ID {
+                    vortex_bail!("metadata should not be among children");
+                }
```
Comment (robert3005) [inline vortex-file/src/read/layouts/chunked.rs]: it seems weird to throw here if we filter it out in `self.children()`

File: vortex-file/src/read/layouts/chunked.rs [hunk 2]
```diff
@@ -77,13 +79,40 @@ impl ChunkedLayout {
...
 
+    #[allow(dead_code)]
+    fn metadata_layout(&self) -> VortexResult<Option<Box<dyn LayoutReader>>> {
+        self.has_metadata()
+            .then(|| {
+                let metadata_fb = self
+                    .flatbuffer()
+                    .children()
+                    .unwrap_or_default()
+                    .iter()
```
Comment (robert3005) [inline vortex-file/src/read/layouts/chunked.rs]: flatbuffer vector derefs to slice so you can replace this with `self
                    .flatbuffer()
                    .children()
                    .ok_or_else(|| )?[0]`

File: vortex-file/src/read/layouts/chunked.rs (hunk 1 above)
Comment (danking) [reply vortex-file/src/read/layouts/chunked.rs]: Removed.

File: vortex-file/src/read/layouts/chunked.rs (hunk 2 above)
Comment (danking) [reply vortex-file/src/read/layouts/chunked.rs]: I had to use `.get(0)` but that worked.

**Review by robert3005 — APPROVED** [review]

**Pull request merged**